### ISSUE TITLE
🐛 [firebase_messaging] v8.0.0-dev.2 onMessage is fired twice

### ISSUE THREAD

**Issue by lazylazyllama** [FirebaseExtended/flutterfire#3991]
## Bug report

Since I updated to firebase_messaging v8.0.0-dev.1 or dev.2 its the same.. `onMessage` is fired twice, when I get a message.

```
D/FLTFireMsgReceiver(22164): broadcast received for message
I/flutter (22164): {data: {moduleId: 113a20f6-08ce-47a5-a85c-b0179b6b862a, type: updateData, title: Test, message: Test Nachricht}, messageId: 0:1603998491174878%06db50d7f9fd7ecd, sentTime: 1603998490818, from: /topics/schulinfoapp_beta-v4_283, ttl: 86400}
I/flutter (22164): onMessage: Instance of 'RemoteMessage'
I/flutter (22164): onMessage: Instance of 'RemoteMessage'
```

### Steps to reproduce

Use the firebase_messaging example with v8.0.0 and receive a message.

### Expected behavior

`onMessage` should only be called once.

**Comment by darshankawar** [comment]
@lazylazyllama 
There's similar open issue describing your case, [1669](https://github.com/FirebaseExtended/flutterfire/issues/1669).
Please follow-up there for updates and any questions you may have.
Closing this as duplicate.

**Comment by lazylazyllama** [comment]
True, but this issue is very old and is about the old firebase_messaging version. Which is weird, because the old version worked perfectly fine for me. I only experienced this issue specifically in the 8.0.0+ versions.

**Comment by Salakar** [comment]
@lazylazyllama which platform are you seeing this behavior on?

**Comment by lazylazyllama** [comment]
Android

**Comment by Salakar** [comment]
Ok thanks, I'll try see if we can reproduce this.

**Comment by Salakar** [comment]
Please can you try `8.0.0-dev.6` - this should now be fixed.

**Comment by lazylazyllama** [comment]
I just tried it, but sadly the issue is still there

**Comment by Salakar** [comment]
Could you share your dart code? Thanks

**Comment by lazylazyllama** [comment]
I'll try to build a reproducible example within the next week. Sadly I'm very busy right now, and the codebase in which I encountered this problem is way to complicated to just upload it here .. considering business logic etc. 

Also while playing around a bit last week I was able to reproduce that this issue does not occur after I stripped down my application down to the very basics, so it's quite a complicated thing to figure out what actually is causing the problem.

But I'll figure it out. I just can't say when I will find time. If this issue gets closed I'll just open a new one, as soon as I have a reproducible example :)

**Comment by pureimpro** [comment]
@lazylazyllama I was facing this issue also, and realized via VSCode widget Inspector, that two instances of the same widgets were present in the widget tree (i.e. my home page), and each one of them was triggered by firebase messages. Maybe you're in a similar situation with your app. I had misconfigured my splash screen page to push my home page on top of my root page (which actually is my home page) and that was my mistake. Hope that helps

**Comment by lazylazyllama** [comment]
@pureimpro actually this is exactly what was going on at my side aswell :O thank you very very much :) You saved quite a lot of time :)

I was using a custom static method called `initializeFcmNotification()`, in which I set up `FirebaseMessaging.onMessage.listen((message) {}`. Because my ui rerendered once, this function was called twice and thus the onMessage body would be called twice aswell on every incoming message.

I fixed it by adding a flag to check, if `initializeFcmNotification()` is called for the first time or not.
